### PR TITLE
[ACTIONS] Mailgun

### PR DESCRIPTION
**Pull request by compwright**

### CODE REVIEW

File: components/mailgun/actions/send-email/send-email.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      description: "Enables or disables DKIM signatures",
+      optional: true,
+    },
+    tracking: {
+      type: "boolean",
+      default: true,
+      description: "Enables or disables tracking",
+      optional: true,
+    },
+    haltOnError: {
```
Comment (dylburger) [inline components/mailgun/actions/send-email/send-email.js:70]: As an aside, this led to a good discussion internally. I think we're likely to make this a step-level primitive (such that you can control the error-handling logic for any step native to the platform). For now this pattern is fine.

File: components/mailgun/actions/send-email/send-email.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      type: "string",
+      label: "HTML",
+      description: "Message body (HTML)",
+    },
+    testMode: {
+      type: "boolean",
+      default: true,
+      description: "Enables sending in test mode",
+    },
+    dkim: {
```
Comment (dylburger) [inline components/mailgun/actions/send-email/send-email.js:57]: For more complex props like this, does it make sense to link to the Mailgun docs in the description for an explanation of how the functionality works? The `description` should support Markdown links.

File: components/mailgun/actions/send-email/send-email.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      type: "string",
+      label: "From Email",
+      description: "Sender email address",
+    },
+    replyTo: {
+      type: "string",
+      label: "Reply Email",
+      description: "Sender Reply-To email address",
+    },
+    to: {
```
Comment (dylburger) [inline components/mailgun/actions/send-email/send-email.js:32]: Should `cc` and `bcc` be supported as optional string arrays, as well?

File: components/mailgun/actions/suppress-email/suppress-email.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      ],
+    },
+  },
+  async run () {
+    try {
+      const suppression = {
+        address: this.email,
+      };
+
+      switch (this.category) {
```
Comment (dylburger) [inline components/mailgun/actions/suppress-email/suppress-email.js:62]: Is a `default` case necessary to capture any type of code / error / tag? Or does the `suppression` object above capture the correct default case?

File: components/mailgun/mailgun.app.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    api (api) {
+      const mailgun = new Mailgun(formData);
+      const mg = mailgun.client({
+        username: "api",
+        key: this.$auth.api_key,
+        public_key: this.$auth.api_key,
+      });
+      return mg[api];
+    },
+    async suppress (domain, type, suppression) {
```
Comment (dylburger) [inline components/mailgun/mailgun.app.js:18]: Is `suppress` not available via the JS client lib?

File: components/mailgun/actions/send-email/send-email.js (hunk 3 above)
Comment (compwright) [reply components/mailgun/actions/send-email/send-email.js:32]: Good catch!

File: components/mailgun/mailgun.app.js (hunk 5 above)
Comment (compwright) [reply components/mailgun/mailgun.app.js:18]: @dylburger heh, it is but no matter what I tried I couldn't get it to work. I think it must have a bug, or Mailgun changed their API behavior and their official client library hasn't caught up. Whatever it is, I noticed that it is particularly touchy about the Content-Type header and the exact format of the request body.

File: components/mailgun/actions/suppress-email/suppress-email.js (hunk 4 above)
Comment (compwright) [reply components/mailgun/actions/suppress-email/suppress-email.js:62]: @dylburger no default case needed, bounces and unsubscribes are the only cases that accept additional information, and all cases require address.

**Comment by compwright** [comment]
@dylburger thanks for the review! Should be good to go.

**Pull request merged**